apiserver/facades/client/storage/storage.go

```diff
@@ -1145,7 +1149,16 @@ func (a *StorageAPI) DeletePool(p params.StoragePoolDeleteArgs) (params.ErrorRes
 	if err := a.checkCanWrite(); err != nil {
 		return results, errors.Trace(err)
 	}
+	poolsInUse, err := a.storagePoolsInUse(p.Pools)
+	if err != nil {
+		return results, errors.Trace(err)
+	}
+
 	for i, pool := range p.Pools {
+		if poolsInUse[pool.Name] {
```

Comment:
This is not really what we want - we want the Delete() call below to return an error if the specific pool being deleted is in use